src/components/ButtonMenu.tsx

```diff
@@ -30,8 +29,11 @@ interface ButtonMenuProps {
   tooltip?: ReactNode;
 }
 
-export function ButtonMenu(props: ButtonMenuProps) {
-  const { trigger, items, placement, persistentItems, defaultOpen, disabled, tooltip } = props;
+interface ButtonMenuDatePickerProps extends ButtonMenuBaseProps, DatePickerProps {}
+interface ButtonMenuItemsProps extends ButtonMenuBaseProps, MenuItemsProps {}
```

Comment:
Not 100% sure I like this approach of using the `DatePickerProps` directly as it is not super clear that doing `<ButtonMenu onSelect={...} />` is going to give you a DatePicker rather than a Menu. I am contemplating instead doing something like:
```
interface ButtonMenuDatePickerProps extends ButtonMenuBaseProps, DatePickerProps {
  as: "datePicker"
}
```
Basically the same thing, but forcing the user to add `as="datePicker"` in the props... not sure if necessary. Open to other ideas.

Reply:
I've not looked at the code at all, but is this better done as a new component? Like ButtonDateMenu? Seems like a pretty specialized use case.

Reply:
It'd basically be a copy & paste of ButtonMenu with a different component shown in the overlay. It could be its own component, but if felt like such a copy and paste that I wanted to improve the existing ButtonMenu to be more flexible about the overlay. This also included a good refactor of the ButtonMenu to remove the Menu specific logic into the Menu component. I'll probably keep that part of the refactoring around.

Reply:
Sounds good. I've updated this PR to create a new `ButtonDatePicker` component and an `OverlayTrigger` component that includes the sharable pieces between ButtonDatePicker and ButtonMenu.